IdentityCore/src/requests/sdk/msal/MSIDDefaultTokenResponseValidator.m

```diff
@@ -50,8 +50,19 @@ - (BOOL)validateTokenResult:(MSIDTokenResult *)tokenResult
 
     NSOrderedSet *grantedScopes = tokenResult.accessToken.scopes;
     NSOrderedSet *normalizedGrantedScopes = grantedScopes.normalizedScopeSet;
+    NSOrderedSet *requestedScopes = configuration.scopes;
+    NSOrderedSet *normalizedRequestedScopes = requestedScopes.normalizedScopeSet;
+    
+    // Include email as scope returned if it was requested and server did not include it.
```

Comment:
why server doesn't return it if it was requested? Should it be fixed on the server side instead?

Reply:
Replied internally. This was an agreement between us and the other teams; there are some cases where the email scope is not included in the granted scopes list, but it is added to the claims in the id_token.